src/main/webapp/app/exercises/file-upload/participate/file-upload-submission.service.ts

```diff
@@ -89,7 +89,14 @@ export class FileUploadSubmissionService {
             params = params.set('lock', 'true');
         }
 
-        return this.http.get<FileUploadSubmission>(url, { params }).pipe(map((res: FileUploadSubmission) => this.submissionService.convertSubmissionFromServer(res)));
+        return this.http.get<FileUploadSubmission>(url, { params }).pipe(
+            map((res?: FileUploadSubmission) => {
+                if (res) {
+                    return this.submissionService.convertSubmissionFromServer(res);
+                }
+                return undefined;
```

Comment:
```suggestion
                if (!res) {
                    return undefined;
                }
                return this.submissionService.convertSubmissionFromServer(res);
```

Just a suggestion (matter of taste), I'll leave it to you